openmls/src/group/core_group/proposals.rs

```diff
@@ -49,6 +49,14 @@ impl ProposalStore {
     pub(crate) fn empty(&mut self) {
         self.queued_proposals = Vec::new();
     }
+    pub(crate) fn remove(&mut self, proposal_ref: ProposalRef) -> Option<()> {
+        let index = self
+            .queued_proposals
+            .iter()
+            .position(|p| p.proposal_reference() == proposal_ref)?;
+        self.queued_proposals.swap_remove(index);
```

Comment:
I'm not sure about this. `swap_remove` changes the order of elements. I think the more safe thing to do would be to just remove the proposal.